### DISCUSSION TITLE
Add a new nodepool did not complete

### DISCUSSION THREAD

**Discussion by cro** [mysticaltech/terraform-hcloud-kube-hetzner#719] [Q&A] [answered]
Greetings,

I was playing with Grafana Cloud which has a neat feature that tells you if your k8s cluster is over or under utilized.  Currently mine is severely underutilized so I thought I would add a new nodepool with smaller machines & save some money.  I added the `agent-medium` section to my kube.tf, intending to later change the `count` for `agent-large` to zero:

```   
{
      name        = "agent-large",
      server_type = "cpx31",
      location    = "ash",
      labels      = [],
      taints      = [],
      count       = 3
    },

    {
      name        = "agent-medium",
      server_type = "cpx21",
      location    = "ash",
      labels      = [],
      taints      = [],
      count       = 3
    }
```

But the `terraform apply` seems to be stuck at `waiting for microOS to become available`.  In the Hetzner UI I see the new machines but they do not have private IPs allocated.  If I connect to their consoles I am presented with a login prompt, so clearly the machines are up and running.  Am I missing something?

**Comment by mysticaltech** [comment]
@cro Make sure to upgrade your project to the latest and greatest version, and that you have a snapshot present. See the readme!

**Comment by cro** [comment, marked as answer]
I was on the latest, and I did have a snapshot.  However, I had never done `hcloud context create <project name>`.  So I renamed my Default project in the console to `hetzner-k3s`, did `hcloud context create hetzner-k3s; hcloud context use hetzner-k3s` and re-ran the terraform apply.  All is well.

This is probably an artifact of how long ago I started with the project.

**Reply by mysticaltech** [reply]
@cro terraform does not use the hcloud-cli, so probably you just needed to re-run terraform, somehow! It happens.
